src/DIPS.Xamarin.UI/Controls/Toast/Toast.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace DIPS.Xamarin.UI.Controls.Toast
+{
+    /// <summary>
+    ///     Toast control that would appear on top of the presented view
+    /// </summary>
+    public static class Toast
+    {
+        private static ToastCore? ToastCore { get; set; }
+
+        internal static void Initialize()
+        {
+            ToastCore?.Dispose();
+            ToastCore = new ToastCore();
+        }
+
+        /// <summary>
+        ///     Display a Toast
+        /// </summary>
+        /// <param name="text">Text to be displayed inside the toast</param>
+        /// <param name="options">An <see cref="Action{ToastOptions}" /> to modify Toast options</param>
+        /// <param name="layout">An <see cref="Action{ToastLayout}" /> to modify Toast layout</param>
+        /// <returns>A void <c>Task</c></returns>
+        public static async Task DisplayToast(string text, Action<ToastOptions> options, Action<ToastLayout> layout)
+        {
+            try
+            {
+                if (ToastCore == null)
+                {
+                    throw new ApplicationException($"Invoke {nameof(Initialize)} first");
```

Comment:
how is the consumer supposed to do Initialize as it is Internal?

Reply:
Scrap code